controllers/metal3.io/host_state_machine.go

```diff
@@ -284,7 +285,18 @@ func (hsm *hostStateMachine) handleReady(info *reconcileInfo) actionResult {
 	actResult := hsm.Reconciler.actionManageReady(hsm.Provisioner, info)
 
 	if _, complete := actResult.(actionComplete); complete {
+		hsm.NextState = metal3v1alpha1.StatePreparing
+	}
+	return actResult
+}
+
+func (hsm *hostStateMachine) handlePreparing(info *reconcileInfo) actionResult {
+	actResult := hsm.Reconciler.actionPreparing(hsm.Provisioner, info)
+	switch actResult.(type) {
+	case actionComplete:
 		hsm.NextState = metal3v1alpha1.StateProvisioning
+	case actionFailed:
+		hsm.NextState = metal3v1alpha1.StateReady
```

Comment:
Why going back to Ready in case of action failure? Wouldn't be enough reusing the existing retry mechanism (based on the ErrorCount) and remain in the same state?